src/main/java/com/thebois/models/inventory/items/Item.java

```diff
@@ -0,0 +1,24 @@
+package com.thebois.models.inventory.items;
+
+/**
+ * A generic item of a specific type.
+ */
+public class Item implements IItem {
+
+    private final ItemType type;
+
+    /**
+     * Instantiates with a given type.
+     *
+     * @param type The type of the item.
+     */
+    public Item(final ItemType type) {
+        this.type = type;
+    }
+
+    @Override
+    public ItemType getType() {
+        return type;
+    }
+
+}
```

Comment:
Is this only to have a factory?

Reply:
I noticed that both `Log` and `Rock` looked exactly the same, except for the type. After I made the refactor to make every item accessed through `IItem`, created by the factory, then it didn't matter anymore to the outside world what the implementing classes were.

So I simplified it to just use this `Item` class. In the future we might have to create specific classes for specific items, but that won't matter as to the outside world they'll all be `IItem`s.